src/app/pledge/components/new-pledge/new-pledge.component.spec.ts

```diff
@@ -189,4 +189,27 @@ describe('NewPledgeComponent', () => {
 
     expect(instance.donationSelection).toBe('custom');
   });
+
+  it('should display the loading spinner', async () => {
+    const { find, instance } = await shallow.render();
+
+    instance.waiting = true;
+
+    expect(find('pr-loading-spinner')).toBeTruthy();
+  });
+
+  it('should display the correct amount of storage if the transaction is succesful', async () => {
+    const { find, instance, fixture } = await shallow.render();
+
+    instance.isSuccessful = true;
+    instance.amountInGb = 5;
+
+    fixture.detectChanges();
+
+    const displayedMessage = find('.success-message');
+
+    expect(displayedMessage.nativeElement.textContent).toContain(
+      'Success! 5 GB of Permanent storage has been added to your account.',
```

Comment:
This test seems a bit too specific, ideally it should only check that `.success-message` exists and that the text content features the amount of GB added. Otherwise if it's too specific we'll end up with a failing test if we want to reword things later.